frontend/lib/location/determine_position.dart

```diff
@@ -19,36 +24,73 @@ Future<Position> determinePosition() async {
 /// When the location services are not enabled or permissions
 /// are denied the `Future` will return with an `PositionNotAvailableException`.
 /// Else it will complete without a value.
-Future<void> requestPermissionToDeterminePosition() async {
+Future<void> requestPermissionToDeterminePosition(
+    {BuildContext userInteract}) async {
   var serviceEnabled = await Geolocator.isLocationServiceEnabled();
   if (!serviceEnabled) {
-    throw PositionNotAvailableException('Location services are disabled.');
+    if (userInteract != null) {
+      await Geolocator.openLocationSettings();
+    } else {
+      throw PositionNotAvailableException('Location services are disabled.');
+    }
   }
 
-  // Geolocator will fail if permissions are requested again before previous
-  // request was finished.
-  await _mutex.protect(() async {
-    var permission = await Geolocator.checkPermission();
-    if (permission == LocationPermission.deniedForever ||
-        _doNotAskForPermissionAgain) {
+  var permission = await Geolocator.checkPermission();
+  if (permission == LocationPermission.deniedForever) {
+    if (userInteract != null) {
+      var result = await showDialog(
+          context: userInteract,
+          builder: (context) => AlertDialog(
+                title: Text("Standortberechtigung freigeben"),
+                content: Text(
+                    "Geben Sie in den App-Einstellungen die"
+                    " Standordberechtigung frei."),
+                actions: [
+                  TextButton(
+                      child: Text("Abbrechen"),
+                      onPressed: () => Navigator.of(context).pop(false)),
+                  TextButton(
+                      child: Text("Okay"),
```

Comment:
Hm, personally I didn't fully read the info text above and just wanted to click "Okay" to get rid of the dialogue and then I was kind of surprised that it took me directly to the settings. Maybe name the button "Einstellungen öffnen" but that might be too long on some smaller devices, idk. Nothing important but I wanted to leave that note here :D